ethcore/src/client/client.rs

```diff
@@ -1652,6 +1652,32 @@ impl MiningBlockChainClient for Client {
 		open_block
 	}
 
+	fn reopen_block(&self, block: ClosedBlock) -> OpenBlock {
```

Comment:
this logic is specifically for the miner, so couldn't it be kept completely within there? client is bloated enough already

 rather than fetching all valid uncles again, the miner could just check if any of the new headers in `imported` is a valid uncle.